banzai_floyds_ui/gui/utils/file_utils.py

```diff
@@ -0,0 +1,41 @@
+from astropy.io import fits
+from django.conf import settings
+import asyncio
+import httpx
+import io
+import requests
+
+
+async def fetch(url, params, headers):
+    async with httpx.AsyncClient() as client:
+        response = await client.get(url, params=params, headers=headers)
+    return response
+
+
+async def fetch_all(archive_header, request_params):
+    tasks = [fetch(settings.ARCHIVE_URL, params, archive_header) for params in request_params]
+    return await asyncio.gather(*tasks)
+
+
+def download_frame(headers, url=f'{settings.ARCHIVE_URL}', params=None, list_endpoint=False):
+    response = requests.get(url, params=params, headers=headers)
+    response.raise_for_status()
+
+    buffer = io.BytesIO()
+    if list_endpoint:
+        buffer.write(requests.get(response.json()['results'][0]['url'], stream=True).content)
```

Comment:
I don't understand what stream=True is doing here? By using .content the whole response to the get() is loaded in to memory anyway. And if I understand it right this is only the url for a fits file so not a massive payload you would want to stream. 

In fact the variable name buffer is a bit misleading - isn't it the fits file path? Maybe rename to something more descriptive?